engine/core/src/main/java/org/datacleaner/job/builder/AbstractComponentBuilder.java

```diff
@@ -472,6 +472,18 @@ private void synchronizeDependentProperties(final ConfiguredPropertyDescriptor p
         }
     }
 
+    private static String getKey(final Object object) {
+        if (object instanceof InputColumn<?>) {
+            final InputColumn<?> inputColumn = (InputColumn<?>) object;
+
+            if (inputColumn.isVirtualColumn()) {
+                return inputColumn.getName();
```

Comment:
A potential danger here is that we don't disallow the same column name from different components, so it's not guaranteed to be unique. OTOH, I can think of lots of other things that can go wrong in that case, so might not really be an issue.